db/migrate/20200802201259_rename_hidden_flags.rb

```diff
@@ -0,0 +1,73 @@
+class RenameHiddenFlags < ActiveRecord::Migration[6.0]
+  def change
+    rename_column :blog_posts, :hidden, :show_on_site
+    change_column_default :blog_posts, :show_on_site, from: false, to: true
+    t = BlogPost.where( show_on_site: true  )
+    f = BlogPost.where( show_on_site: false )
+    t.update( show_on_site: false )
+    f.update( show_on_site: true  )
```

Comment:
Are you able to check if these migration lines are doing what's intended?

I've a feeling the following will happen:

```rb
# t and f are lazy relations, the queries will not run until the `update_all` calls.
t = BlogPost.where( show_on_site: true  )
f = BlogPost.where( show_on_site: false )

# NB. Assuming these updates should use `update_all` instead of `update`.

# 1. Runs `UPDATE blog_posts SET show_on_site = 'f' WHERE show_on_site = 't';
t.update_all( show_on_site: false )

# 2. At this point all records in blog_post now have show_on_site == false

# 3. Runs `UPDATE blog_posts SET show_on_site = 't' WHERE show_on_site = 'f';
f.update_all( show_on_site: true  )

# 4. Now all blog_posts have show_on_site == true
```

Reply:
Argh. Yes, you're correct I'm sure - I keep forgetting about the lazy thing :) How can I nail down the items collected in t and f before I run the updates?

Reply:
Try running a single update statement that flips the values using the `NOT` operator:

```rb
execute "UPDATE blog_posts SET show_on_site = NOT show_on_site"
```